server.rb

```diff
@@ -37,26 +43,67 @@ def language(id)
       sender_id = messaging["sender"]["id"]
       text = messaging["message"]["text"]
 
-      unless @@users[sender_id]
-        Bot.new.send_message(sender_id, "To what language would you like me to translate? I know a lot")
-        Bot.new.send_message(sender_id, "Please respond with '/language'. For example, '/English'")
-      end
-
-      @@users[sender_id] ||= "English"
-
-      if text[0] == "/"
-        @@users[sender_id] = text[1..-1]
-      end
+      # unless @@users[sender_id]
+      #   Bot.new.send_message(sender_id, "To what language would you like me to translate? I know a lot")
+      #   Bot.new.send_message(sender_id, "Please respond with '/language'. For example, '/English'")
+      # end
+      #
+      # @@users[sender_id] ||= {}
+      # @@users[sender_id][:language] ||= "English"
+      # @@users[sender_id][:conversationType] ||= 'initial'
+      #
+      # if text[0] == "/"
+      #   @@users[sender_id][:language] = text[1..-1]
+      # end
+      #
+      # ap @@users
+      # choice = language(sender_id)
+      #
+      # if choice == nil
+      #   Bot.new.send_message(sender_id, "Sorry couldn't find #{@@users[sender_id][:language]} so I'm going to speak English for now.")
+      #   Bot.new.send_message(sender_id, "Please try another language with '/language'. For example, '/English' for translations to English")
+      # else
+      #   reply = EasyTranslate.translate(text, to: choice)
+      #   Bot.new.send_message(sender_id, reply)
+      # end
+      @@users[sender_id] ||= {}
+      @@users[sender_id][:language] ||= "English"
+      @@users[sender_id][:conversationType] ||= 'initial'
+      conversationType = @@users[sender_id][:conversationType]
+        case conversationType
+          when 'initial'
+            @@users[sender_id][:conversationType] = 'begun'
+            Bot.new.send_message(sender_id, "To what language would you like me to translate? I know a lot")
+            Bot.new.send_message(sender_id, "Please respond with '/language'. For example, '/English'")
```

Comment:
I wanted them to be able to set the language at any time `/German` instead of having to do **2 actions** `/language` then `German`